src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -27,6 +51,7 @@ public OssSecretMasker(IEnumerable<RegexPattern> patterns)
     private OssSecretMasker(OssSecretMasker copy)
     {
         _secretMasker = copy._secretMasker.Clone();
+        _telemetry = copy._telemetry?.Clone();
```

Comment:
>_telemetry [](http://example.com/codeflow?start=26&length=10)

Our threading/other guarantees are non-obvious to me now and I don't have a good mental model of what's going on and whether our code is in a sensible, clean condition. Maybe you have a clearer picture?

We have synchronization in the masker that implies multithreaded access. I will note that our clone operation here doesn't account for the possibility that the copied secret masker will receive/process telemetry between the secret masker clone and the telemetry clone. Possibly putting our cloned data out of sync, yes? 

maybe we should use the copy._secretMasker.SyncObject if we can to coordinate things.

Reply:
This is fixed. At great cost. Requires https://github.com/microsoft/security-utilities/pull/208 in draft as well.

We no longer support clone in the library and the sole use of clone in the agent is removed.